test/NuGet.Core.Tests/NuGet.ProjectModel.Test/PackageSpecWriterTests.cs

```diff
@@ -725,6 +727,59 @@ public void Write_RestoreAuditProperties_RoundTrips()
             VerifyJsonPackageSpecRoundTrip(json);
         }
 
+        [Fact]
+        public void RestoreMetadataWithMacros_RoundTrips()
+        {
+            // Arrange
+            var json = @"{  
+                            ""restore"": {
+    ""projectUniqueName"": ""C:\\users\\me\\source\\code\\project.csproj"",
+    ""projectName"": ""project"",
+    ""projectPath"": ""C:\\users\\me\\source\\code\\project.csproj"",
+    ""projectJsonPath"": ""C:\\users\\me\\source\\code\\project.json"",
+    ""packagesPath"": ""$(User).nuget\\packages"",
+    ""outputPath"": ""C:\\users\\me\\source\\code\\obj"",
```

Comment:
same question as another comment I wrote. All these `c:\users\me` paths seem like they should be replaced with `$(User)`, like it is for `packagePaths`

Reply:
In order to completely make the assets file able to float from machine to machine, we'd need to apply macro to *every* path, and that'd include project refs etc. So doing these without that is not going far enough. 

The reason why I did the nuget.config and fallback folders is because those are defaults, things you can't control. For example, the nuget.config will get created whether you want it or not. 

I chose to go the more minimal direction for the time being as I didn't want to pay the cost for now.

Reply:
Makes sense. And yes, thinking about it now, these are the paths to the project, which won't change per-user, so it's logical that it doesn't need to use the macros 👍